numpy/random/_generator.pyi

```diff
@@ -92,17 +87,31 @@ class Generator:
     def standard_normal(  # type: ignore[misc]
         self,
         size: _ShapeLike = ...,
-        dtype: Union[_DTypeLikeFloat32, _DTypeLikeFloat64] = ...,
-        out: Optional[ndarray[Any, dtype[Union[float32, float64]]]] = ...,
-    ) -> ndarray[Any, dtype[Union[float32, float64]]]: ...
+    ) -> ndarray[Any, dtype[float64]]: ...
     @overload
-    def permutation(self, x: int, axis: int = ...) -> ndarray[Any, dtype[int64]]: ...
+    def standard_normal(  # type: ignore[misc]
+        self,
+        *,
+        out: ndarray[Any, dtype[float64]] = ...,
+    ) -> ndarray[Any, dtype[float64]]: ...
```

Comment:
Is that OK here since out has to be ` ndarray[Any, dtype[float64]]` and can't be any other type?

Reply:
To be specific

In these 3:

1. When using `out` and not type, `out` must be a `float64` array.
2. When using `dtype=float32` out can be `None` or a `float32` array, but nothing else.
3. When using `dtype=float64` out can be `None` or a `float64` array, but nothing else.

Can I simplify these? Because of the `dtype` keyword argument that defaults to `np.float64`, it isn't a case of getting the type from `out` when `dtype` is missing.  Would have been a good design to do this, but I think that boat has sailed.

```
    @overload
    def standard_normal(  # type: ignore[misc]
        self,
        *,
        out: ndarray[Any, dtype[float64]] = ...,
    ) -> ndarray[Any, dtype[float64]]: ...
    @overload
    def standard_normal(  # type: ignore[misc]
        self,
        size: _ShapeLike = ...,
        dtype: _DTypeLikeFloat32 = ...,
        out: Optional[ndarray[Any, dtype[float32]]] = ...,
    ) -> ndarray[Any, dtype[float32]]: ...
    @overload
    def standard_normal(  # type: ignore[misc]
        self,
        size: _ShapeLike = ...,
        dtype: _DTypeLikeFloat64 = ...,
        out: Optional[ndarray[Any, dtype[float64]]] = ...,
    ) -> ndarray[Any, dtype[float64]]: ...
```

Reply:
Considering the limited set of accepted dtypes I don't think you can actually simplify this any further.
Let's just keep things as they are.